docs/index.rst

```diff
@@ -33,33 +28,32 @@ MSAL proposes a clean separation between
 They are implemented as two separated classes,
 with different methods for different authentication scenarios.
 
-PublicClientApplication
+ClientApplication
```

Comment:
The pre-existing section header (now changed to "API", which is fine), and the two paragraphs above, were trying to promote the 2 (not 3) classes as our conceptual API. Given that you now bring in the `:inherited-members` trick (Yay!), I think we can simply NOT mention parent class `ClientApplication` at all.

Reply:
Removed that class